### PR TITLE
Integrate minimal CRDT (Grow-only Set)

### PR DESCRIPTION
**Pull request by RandoomWalks**
• Created GSet.js with a class for managing a set of items
 • Implemented IndexedDBUtil.js for IndexedDB operations
 • Developed dbWorker.js as a web worker to handle database actions
 • Set up main.js to interact with the web worker
 • Wrote TypeScript versions of GSet, IndexedDBUtil, and dbWorker
 • Configured TypeScript compilation settings in tsconfig.json and tsconfig.worker.json

### CODE REVIEW

File: tsconfig.worker.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
+{
+    "compilerOptions": {
+        "target": "es6",
+        "module": "none", // Web workers do not support ES modules
```
Comment (hwrdtm) [inline tsconfig.worker.json]: you can try using `.mts` file extension to tell the tsc compiler that it should use commonjs

File: dist/GSet.js
Comment (hwrdtm) [inline dist/GSet.js:1]: Let's gitignore the `dist` directories.

File: src/IndexedDBUtil.ts [hunk 2]
```diff
@@ -1,51 +1,58 @@
...
+    private db: IDBDatabase | null = null; // Reference to the IndexedDB database.
 
+    // Initialize the database connection.
     constructor() {
         this.openDB();
     }
 
+    // Opens the database, creating it if it doesn't exist.
     private openDB(): void {
         const request = indexedDB.open("SynchronizationDB", 1);
```
Comment (hwrdtm) [inline src/IndexedDBUtil.ts]: is this using a client pool under the hood? if not, is it worth us setting up a client pool?

Comment (hwrdtm) [reply src/IndexedDBUtil.ts]: also we talked about making sure we minimize the section of the critical section as much as pos.

Comment (RandoomWalks) [reply src/IndexedDBUtil.ts]: looking into this

File: src/worker/dbWorker.ts [hunk 3]
```diff
@@ -13,58 +14,24 @@ interface WorkerMessage {
...
-}
-
 // Function to add an item to a GSet and store it in IndexedDB
-function addToSet(db: IDBDatabase, id: string, item: any): Promise<void> {
+async function addToSet(id: string, item: any): Promise<void> {
     let gSet = sets.get(id) || new GSet<any>();
     gSet.add(item);
     sets.set(id, gSet);
-    return storeSet(db, id, gSet);
+    await dbUtil.addSet(id, gSet);
```
Comment (hwrdtm) [inline src/worker/dbWorker.ts:25]: The `await` is redundant

```suggestion
    dbUtil.addSet(id, gSet);
```

File: src/worker/dbWorker.ts [hunk 4]
```diff
@@ -73,54 +40,37 @@ function getItems(id: string): any[] | undefined {
...
-        if (!gDB) {
-            throw new Error('Database has not been initialized.');
+// Event handler for messages received from the main thread
+self.onmessage = async (e: MessageEvent<WorkerMessage>) => {
+    const { type, payload } = e.data;
+    try {
+        switch (type) {
+            case 'open':
+                IndexedDBUtil.getInstance();
+                self.postMessage({ type: 'opened' });
```
Comment (hwrdtm) [inline src/worker/dbWorker.ts:50]: would be nice to use enums or union of string types.

**Pull request closed without merging**